datasource/ethereum/src/ethereum_adapter.rs

```diff
@@ -575,26 +648,160 @@ where
         )
     }
 
-    fn find_first_blocks_with_logs(
+    fn calls_in_block(
+        &self,
+        logger: &Logger,
+        block_number: u64,
+        block_hash: H256,
+    ) -> Box<Future<Item = Vec<EthereumCall>, Error = Error> + Send> {
+        let eth = self.clone();
+        let call_filter = EthereumCallFilter {
+            contract_addresses_function_signatures: HashMap::new(),
+        };
+        let calls = eth.call_stream(&logger, block_number, block_number, call_filter)
+            .collect()
+            .map(|call_chunks| match call_chunks.len() {
+                0 => vec![],
+                _ => {
+                    call_chunks
+                        .into_iter()
+                        .flatten()
+                        .collect()
+                }
+            })
+            .and_then(move |calls| {
+                // Ensure the `EthereumCall` objects are for the correct Ethereum block
+                // by checking the call's block hash against the desired block hash.
+                // 
+                // DISCLAIMER: If the call stream returns calls for the correct block_number but the
+                // incorrect block_hash we can catch this and error. But if the call streams returns no calls
+                // for the correct block_number but the incorrect block_hash, there is no way to tell.
+                // Since we are asking for all the calls in the block this is not a problem.
```

Comment:
Is this not a problem because a block will always have at least one call? Maybe add an assert `len > 0`. Or is there another reason?

Reply:
Hey Tim! Yeah, that was the assumption. I added the `assert` you mentioned for now, but thinking through this again, I'm not sure that assumption holds.

TODO: Can there be a block without calls? If so a subgraph will stall at that block with this logic.